src/util/tempdir.cpp

```diff
@@ -34,26 +34,30 @@ std::string get_temporary_directory(const std::string &name_template)
   std::string result;
 
   #ifdef _WIN32
-    DWORD dwBufSize = MAX_PATH;
-    char lpPathBuffer[MAX_PATH];
+    DWORD dwBufSize = MAX_PATH+1;
+    char lpPathBuffer[MAX_PATH+1];
     DWORD dwRetVal = GetTempPathA(dwBufSize, lpPathBuffer);
 
     if(dwRetVal > dwBufSize || (dwRetVal == 0))
       throw "GetTempPath failed"; // NOLINT(readability/throw)
 
-    char t[MAX_PATH];
+    // GetTempFileNameA produces <path>\<pre><uuuu>.TMP
+    // where <pre> = "TLO"
+    // Thus, we must make the buffer 1+3+4+1+3=12 characters longer.
 
-    strncpy(t, name_template.c_str(), MAX_PATH);
+    std::size_t size=strlen(lpPathBuffer)+1;
+    std::vector<char> t(lpPathBuffer, lpPathBuffer+size);
+    t.resize(size+12);
```

Comment:
Isn't the pre-existing contents of `t` just going to be ignored and overwritten? In my opinion it would suffice to declare another array of size `MAX_PATH` (no `+1` required according to the documentation).

Reply:
ok, back to string buffer